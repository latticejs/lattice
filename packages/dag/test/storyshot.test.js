import initStoryshots, { snapshotWithOptions } from '@storybook/addon-storyshots';

function createNodeMock(element) {
  if (element.type === 'text') {
    return {
      getBoundingClientRect() {
        return {};
      }
    };
  }
  return null;
}

initStoryshots({
  test: snapshotWithOptions({
    createNodeMock
  })
});
