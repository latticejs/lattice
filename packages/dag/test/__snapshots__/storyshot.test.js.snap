// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots dag/advanced editable mode 1`] = `
<div
  contentEditable={true}
  onKeyUp={[Function]}
  style={
    Object {
      "outline": "none",
      "position": "relative",
    }
  }
  suppressContentEditableWarning={true}
>
  <svg
    className="dag-wrapper Dag-root-01"
    height={500}
    onDoubleClick={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    width={500}
  >
    <g
      className="dag__graph"
    >
      <g
        className="dag__edge Dag-dagEdge-04 Dag-dagEditable-05"
        onClick={[Function]}
      >
        <line />
        <polygon
          className="dag__edge-arrow"
          id="dag__line-0"
        />
      </g>
      <g
        className="dag__edge Dag-dagEdge-04 Dag-dagEditable-05"
        onClick={[Function]}
      >
        <line />
        <polygon
          className="dag__edge-arrow"
          id="dag__line-1"
        />
      </g>
      <g
        className="dag__edge Dag-dagEdge-04 Dag-dagEditable-05"
        onClick={[Function]}
      >
        <line />
        <polygon
          className="dag__edge-arrow"
          id="dag__line-2"
        />
      </g>
      <g
        className="dag__edge Dag-dagEdge-04 Dag-dagEditable-05"
        onClick={[Function]}
      >
        <line />
        <polygon
          className="dag__edge-arrow"
          id="dag__line-3"
        />
      </g>
      <g
        className="dag__edge Dag-dagEdge-04 Dag-dagEditable-05"
        onClick={[Function]}
      >
        <line />
        <polygon
          className="dag__edge-arrow"
          id="dag__line-4"
        />
      </g>
      <g
        className="dag__edge Dag-dagEdge-04 Dag-dagEditable-05"
        onClick={[Function]}
      >
        <line />
        <polygon
          className="dag__edge-arrow"
          id="dag__line-5"
        />
      </g>
      <g
        className="dag__node Dag-dagNode-02 Dag-dagEditable-05"
        id="dag__node--app"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        <circle />
        <text
          className="Dag-dagNodeText-03"
        />
      </g>
      <g
        className="dag__node Dag-dagNode-02 Dag-dagEditable-05"
        id="dag__node--lodash"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        <circle />
        <text
          className="Dag-dagNodeText-03"
        />
      </g>
      <g
        className="dag__node Dag-dagNode-02 Dag-dagEditable-05"
        id="dag__node--react"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        <circle />
        <text
          className="Dag-dagNodeText-03"
        />
      </g>
      <g
        className="dag__node Dag-dagNode-02 Dag-dagEditable-05"
        id="dag__node--react-dom"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        <circle />
        <text
          className="Dag-dagNodeText-03"
        />
      </g>
      <g
        className="dag__node Dag-dagNode-02 Dag-dagEditable-05"
        id="dag__node--apollo"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        <circle />
        <text
          className="Dag-dagNodeText-03"
        />
      </g>
      <g
        className="dag__node Dag-dagNode-02 Dag-dagEditable-05"
        id="dag__node--enzyme"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        <circle />
        <text
          className="Dag-dagNodeText-03"
        />
      </g>
    </g>
  </svg>
</div>
`;

exports[`Storyshots dag/basic no wrapper 1`] = `
<div
  contentEditable={true}
  onKeyUp={[Function]}
  style={
    Object {
      "outline": "none",
      "position": "relative",
    }
  }
  suppressContentEditableWarning={true}
>
  <svg
    className="dag-wrapper Dag-root"
    height={500}
    onDoubleClick={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    width={500}
  >
    <g
      className="dag__graph"
    >
      <g
        className="dag__edge Dag-dagEdge"
        onClick={[Function]}
      >
        <line />
        <polygon
          className="dag__edge-arrow"
          id="dag__line-0"
        />
      </g>
      <g
        className="dag__edge Dag-dagEdge"
        onClick={[Function]}
      >
        <line />
        <polygon
          className="dag__edge-arrow"
          id="dag__line-1"
        />
      </g>
      <g
        className="dag__edge Dag-dagEdge"
        onClick={[Function]}
      >
        <line />
        <polygon
          className="dag__edge-arrow"
          id="dag__line-2"
        />
      </g>
      <g
        className="dag__edge Dag-dagEdge"
        onClick={[Function]}
      >
        <line />
        <polygon
          className="dag__edge-arrow"
          id="dag__line-3"
        />
      </g>
      <g
        className="dag__edge Dag-dagEdge"
        onClick={[Function]}
      >
        <line />
        <polygon
          className="dag__edge-arrow"
          id="dag__line-4"
        />
      </g>
      <g
        className="dag__edge Dag-dagEdge"
        onClick={[Function]}
      >
        <line />
        <polygon
          className="dag__edge-arrow"
          id="dag__line-5"
        />
      </g>
      <g
        className="dag__node Dag-dagNode"
        id="dag__node--app"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        <circle />
        <text
          className="Dag-dagNodeText"
        />
      </g>
      <g
        className="dag__node Dag-dagNode"
        id="dag__node--lodash"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        <circle />
        <text
          className="Dag-dagNodeText"
        />
      </g>
      <g
        className="dag__node Dag-dagNode"
        id="dag__node--react"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        <circle />
        <text
          className="Dag-dagNodeText"
        />
      </g>
      <g
        className="dag__node Dag-dagNode"
        id="dag__node--react-dom"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        <circle />
        <text
          className="Dag-dagNodeText"
        />
      </g>
      <g
        className="dag__node Dag-dagNode"
        id="dag__node--apollo"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        <circle />
        <text
          className="Dag-dagNodeText"
        />
      </g>
      <g
        className="dag__node Dag-dagNode"
        id="dag__node--enzyme"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        <circle />
        <text
          className="Dag-dagNodeText"
        />
      </g>
    </g>
  </svg>
</div>
`;

exports[`Storyshots dag/themed paper wrapper (dark) 1`] = `
<div
  style={
    Object {
      "height": "100vh",
      "width": "100vw",
    }
  }
>
  <div
    className="MuiPaper-root MuiPaper-elevation2 MuiPaper-rounded"
    style={
      Object {
        "height": "500px",
        "width": "500px",
      }
    }
  >
    <div
      contentEditable={true}
      onKeyUp={[Function]}
      style={
        Object {
          "outline": "none",
          "position": "relative",
        }
      }
      suppressContentEditableWarning={true}
    >
      <svg
        className="dag-wrapper Dag-root"
        height={500}
        onDoubleClick={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
        width={500}
      >
        <g
          className="dag__graph"
        >
          <g
            className="dag__edge Dag-dagEdge"
            onClick={[Function]}
          >
            <line />
            <polygon
              className="dag__edge-arrow"
              id="dag__line-0"
            />
          </g>
          <g
            className="dag__edge Dag-dagEdge"
            onClick={[Function]}
          >
            <line />
            <polygon
              className="dag__edge-arrow"
              id="dag__line-1"
            />
          </g>
          <g
            className="dag__edge Dag-dagEdge"
            onClick={[Function]}
          >
            <line />
            <polygon
              className="dag__edge-arrow"
              id="dag__line-2"
            />
          </g>
          <g
            className="dag__edge Dag-dagEdge"
            onClick={[Function]}
          >
            <line />
            <polygon
              className="dag__edge-arrow"
              id="dag__line-3"
            />
          </g>
          <g
            className="dag__edge Dag-dagEdge"
            onClick={[Function]}
          >
            <line />
            <polygon
              className="dag__edge-arrow"
              id="dag__line-4"
            />
          </g>
          <g
            className="dag__edge Dag-dagEdge"
            onClick={[Function]}
          >
            <line />
            <polygon
              className="dag__edge-arrow"
              id="dag__line-5"
            />
          </g>
          <g
            className="dag__node Dag-dagNode"
            id="dag__node--app"
            onClick={[Function]}
            onDoubleClick={[Function]}
          >
            <circle />
            <text
              className="Dag-dagNodeText"
            />
          </g>
          <g
            className="dag__node Dag-dagNode"
            id="dag__node--lodash"
            onClick={[Function]}
            onDoubleClick={[Function]}
          >
            <circle />
            <text
              className="Dag-dagNodeText"
            />
          </g>
          <g
            className="dag__node Dag-dagNode"
            id="dag__node--react"
            onClick={[Function]}
            onDoubleClick={[Function]}
          >
            <circle />
            <text
              className="Dag-dagNodeText"
            />
          </g>
          <g
            className="dag__node Dag-dagNode"
            id="dag__node--react-dom"
            onClick={[Function]}
            onDoubleClick={[Function]}
          >
            <circle />
            <text
              className="Dag-dagNodeText"
            />
          </g>
          <g
            className="dag__node Dag-dagNode"
            id="dag__node--apollo"
            onClick={[Function]}
            onDoubleClick={[Function]}
          >
            <circle />
            <text
              className="Dag-dagNodeText"
            />
          </g>
          <g
            className="dag__node Dag-dagNode"
            id="dag__node--enzyme"
            onClick={[Function]}
            onDoubleClick={[Function]}
          >
            <circle />
            <text
              className="Dag-dagNodeText"
            />
          </g>
        </g>
      </svg>
    </div>
  </div>
</div>
`;

exports[`Storyshots dag/themed paper wrapper (light) 1`] = `
<div
  style={
    Object {
      "height": "100vh",
      "width": "100vw",
    }
  }
>
  <div
    className="MuiPaper-root MuiPaper-elevation2 MuiPaper-rounded"
    style={
      Object {
        "height": "500px",
        "width": "500px",
      }
    }
  >
    <div
      contentEditable={true}
      onKeyUp={[Function]}
      style={
        Object {
          "outline": "none",
          "position": "relative",
        }
      }
      suppressContentEditableWarning={true}
    >
      <svg
        className="dag-wrapper Dag-root"
        height={500}
        onDoubleClick={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
        width={500}
      >
        <g
          className="dag__graph"
        >
          <g
            className="dag__edge Dag-dagEdge"
            onClick={[Function]}
          >
            <line />
            <polygon
              className="dag__edge-arrow"
              id="dag__line-0"
            />
          </g>
          <g
            className="dag__edge Dag-dagEdge"
            onClick={[Function]}
          >
            <line />
            <polygon
              className="dag__edge-arrow"
              id="dag__line-1"
            />
          </g>
          <g
            className="dag__edge Dag-dagEdge"
            onClick={[Function]}
          >
            <line />
            <polygon
              className="dag__edge-arrow"
              id="dag__line-2"
            />
          </g>
          <g
            className="dag__edge Dag-dagEdge"
            onClick={[Function]}
          >
            <line />
            <polygon
              className="dag__edge-arrow"
              id="dag__line-3"
            />
          </g>
          <g
            className="dag__edge Dag-dagEdge"
            onClick={[Function]}
          >
            <line />
            <polygon
              className="dag__edge-arrow"
              id="dag__line-4"
            />
          </g>
          <g
            className="dag__edge Dag-dagEdge"
            onClick={[Function]}
          >
            <line />
            <polygon
              className="dag__edge-arrow"
              id="dag__line-5"
            />
          </g>
          <g
            className="dag__node Dag-dagNode"
            id="dag__node--app"
            onClick={[Function]}
            onDoubleClick={[Function]}
          >
            <circle />
            <text
              className="Dag-dagNodeText"
            />
          </g>
          <g
            className="dag__node Dag-dagNode"
            id="dag__node--lodash"
            onClick={[Function]}
            onDoubleClick={[Function]}
          >
            <circle />
            <text
              className="Dag-dagNodeText"
            />
          </g>
          <g
            className="dag__node Dag-dagNode"
            id="dag__node--react"
            onClick={[Function]}
            onDoubleClick={[Function]}
          >
            <circle />
            <text
              className="Dag-dagNodeText"
            />
          </g>
          <g
            className="dag__node Dag-dagNode"
            id="dag__node--react-dom"
            onClick={[Function]}
            onDoubleClick={[Function]}
          >
            <circle />
            <text
              className="Dag-dagNodeText"
            />
          </g>
          <g
            className="dag__node Dag-dagNode"
            id="dag__node--apollo"
            onClick={[Function]}
            onDoubleClick={[Function]}
          >
            <circle />
            <text
              className="Dag-dagNodeText"
            />
          </g>
          <g
            className="dag__node Dag-dagNode"
            id="dag__node--enzyme"
            onClick={[Function]}
            onDoubleClick={[Function]}
          >
            <circle />
            <text
              className="Dag-dagNodeText"
            />
          </g>
        </g>
      </svg>
    </div>
  </div>
</div>
`;
